### PR TITLE
Update WizardComponent.php

### PR DESCRIPTION
**Pull request by mlingureanu**
test type for $proceed

### CODE REVIEW

**Comment by ProLoser** [comment]
I'll leave this up to @bancer who's become far more active on this repo than I.

**Comment by ProLoser** [comment]
@mlingureanu since this is your first contribution, can I ask you what this PR is trying to accomplish? What was the problem you ran into that made this fix necessary?

**Comment by mlingureanu** [comment]
The documentation says 

> The process callback must return either true or false. If true, the wizard will continue to the next step; if false, the user will remain on the step and any validation errors will be presented.

I run into trouble, on unit tests, when instead of returning boolean, I need to force a redirect like
return $this->redirect($url);
The problem is, this process callback will return an mocked object in unit tests, and variable $proceed from wizard component will get this mocked object instead of boolean value. 
Next condition `if ($proceed)` without checking type, will assume this object is true and will execute unwanted code.

**Comment by ProLoser** [comment]
Why are you getting an object? Sounds like a bug in your tests.

**Comment by mlingureanu** [comment]
> When testing actions that contain redirect() and other code following the redirect it is generally a good idea to return when redirecting. The reason for this, is that redirect() is mocked in testing, and does not exit like normal. And instead of your code exiting, it will continue to run code following the redirect.

https://book.cakephp.org/2.0/en/development/testing.html#testing-controllers

**Comment by bancer** [comment]
I am not convinced this change is needed while the same can be achieved by `$this->redirect($url); return false;` or `$this->redirect($url); return true;` in the controller. @ProLoser do you have an opinion on that?

**Comment by ProLoser** [comment]
I'm too removed from the code to know how it works anymore, I just don't like strict comparisons to true unless you handle other truthy values differently (which this code doesn't)

File: Controller/Component/WizardComponent.php [hunk 1]
```diff
@@ -382,7 +382,7 @@ public function process($step) {
 					} else {
 						throw new NotImplementedException(sprintf(__('Process Callback not found. Please create Controller::%s', $processCallback)));
 					}
-					if ($proceed) {
+					if ($proceed === true) {
```
Comment (mlingureanu) [inline Controller/Component/WizardComponent.php]: $proceed from line 379 and line 381 should return always boolean, that's why it's a matter of best practices, to test type when dealing with booleans

Comment (mlingureanu) [inline Controller/Component/WizardComponent.php]: https://phpbestpractices.org/#checking-for-null

Comment (mlingureanu) [inline Controller/Component/WizardComponent.php]: Gotchas
When testing the return value of a function that can return either 0 or boolean false, like strpos(), always use === and !==, or you’ll run in to problems.

Comment (ProLoser) [inline Controller/Component/WizardComponent.php]: That specifically talks about falsey values if they represent different things than null.

Again, I think something is wrong with your unit test. I need more context on how this code is called.

**Comment by ProLoser** [comment]
Why does `processCallback` not return a boolean?

**Comment by mlingureanu** [comment]
in WizardComponentTest.php at line 119 

> 	public function redirect($url = null, $status = null, $exit = true) {
> 		// Do not allow redirect() to exit in unit tests.
> 		return parent::redirect($url, $status, false);
> 	}
by the way this comment `// Do not allow redirect() to exit in unit tests.` it's wrong because in unit tests redirect is mocked and third param $exit is false

it's override default cakephp redirect method to not exit and to continue to how Bancer says  

> $this->redirect($url); return false; or $this->redirect($url); return true

this is a snippet from my code:

```
public function processRestaurant() {
		if (!empty($this->request->data)) {
		...
		// Create the redirect url with the new restaurant and the extra values from before.
		$url = array(
			'controller' => 'book',
			'action' => 'table',
			$stepToGo,
			$restaurantId,
		);
		$url = array_merge($url, $extra);
		$this->redirect($url, null, false);
		return false;
	}
	return false;
}
```
I need this redirect, because I need to change param $restaurantId and method redirect from wizard don't allow params to be changed.

**Comment by mlingureanu** [comment]
by the way this comment `// Do not allow redirect() to exit in unit tests.` it's wrong because in unit tests redirect is mocked and third param $exit is false

**Comment by ProLoser** [comment]
I mean I'll leave it to bancer

**Pull request merged**